File: game.cc
#include "game.hh"
#include "raylib.h"

game game::game_init() {
    game g = {};

    g.block = block::block_init(500, 300);

    return g;
}

void game::start() {

}

void game::update() {
    if (IsKeyDown(KEY_SPACE)) {
        x++;
    }
}

void game::draw() {
    ClearBackground(WHITE);

    block.draw();
}

File: goblocks/game.go
package main

import rl "github.com/gen2brain/raylib-go/raylib"

type Game struct {
	Block *Blocks
}

func newGame() *Game {
	return &Game{
		Block: NewBlocks(10),
	}
}

func (g *Game) init() {
	g.Block.Init()
}

func (g *Game) update() {
	g.Block.Update()
}

func (g *Game) draw() {
	rl.ClearBackground(rl.Black)

	g.Block.Draw()
}
